src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/armor/SlimefunArmorPiece.java

```diff
@@ -1,16 +1,19 @@
 package io.github.thebusybiscuit.slimefun4.implementation.items.armor;
 
+import org.bukkit.NamespacedKey;
 import org.bukkit.entity.Player;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.potion.PotionEffect;
 
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
 import me.mrCookieSlime.Slimefun.Lists.RecipeType;
 import me.mrCookieSlime.Slimefun.Objects.Category;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
 import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
 
 public class SlimefunArmorPiece extends SlimefunItem {
 
+    private NamespacedKey id = null;
```

Comment:
This being null here isn't really the best ideas.
I mean, we only use this key for the CustomProtection armor, so it might make sense to just have the getter in that interface and force the developer to implement this field in their implementation class instead.

This way we avoid unnecessary null fields, since those aren't needed in this class here.